app/ota_client_stub.py

```diff
@@ -15,23 +18,55 @@ def __init__(self):
         self._ecu_info = EcuInfo()
         self._ota_client_call = OtaClientCall("50051")
 
-    def update(self, request):
-        response = []
+        # dispatch the requested operations to threadpool
+        self._executor = ThreadPoolExecutor()
+        # a dict to hold the future for each requests if needed
+        self._future = dict()
+
+    def terminate(self):
+        self._executor.shutdown()
```

Comment:
It is better call shutdown in destructor, since _executor is created in constructor.
```suggestion
    def __del__(self):
        self._executor.shutdown()
```